src/main/java/com/patternpedia/api/rest/controller/CandidateController.java

```diff
@@ -0,0 +1,148 @@
+//                        getPatternViewLinks(patternView)))
+                .collect(Collectors.toList());
+
+        return new CollectionModel<>(candidates);
+    }
+
+    @GetMapping(value = "/getById/{candidateId}")
+    @PreAuthorize(value = "#oauth2.hasScope('read')")
+    Candidate getCandidateById(@PathVariable UUID candidateId) {
+        return this.candidateService.getCandidateById(candidateId);
+    }
+
+    @GetMapping(value = "/getByUri/{candidateUri}")
+    Candidate getCandidateByUri(@PathVariable String candidateUri) {
+        return this.candidateService.getCandidateByURI(candidateUri);
+    }
+
+    /**
+     * CREATE Methods
+     */
+    @PostMapping(value = "/create")
+//    @PostMapping(value = "/create")
+//    @PreAuthorize(value = "#oauth2.hasScope('write')")
+//    @CrossOrigin(exposedHeaders = "Location")
+    @ResponseStatus(HttpStatus.CREATED)
+    Candidate newCandidate(@RequestBody CandidateModel candidate) {
+        return this.candidateService.createCandidate(candidate);
+    }
+
+    @PostMapping(value = "/createComment/{candidateId}&{userId}")
+//    @PreAuthorize(value = "#oauth2.hasScope('write')")
+//    @CrossOrigin(exposedHeaders = "Location")
+    @ResponseStatus(HttpStatus.CREATED)
+    Candidate newCandidateComment(@PathVariable UUID candidateId, @PathVariable UUID userId, @RequestBody CandidateComment candidateComment) {
+        return this.candidateService.createComment(candidateId, userId, candidateComment);
+    }
+
+//    @PostMapping(value = "/createFromIssue/{patternLanguageId}")
+////    @PreAuthorize(value = "#oauth2.hasScope('write')")
+////    @CrossOrigin(exposedHeaders = "Location")
+//    @ResponseStatus(HttpStatus.CREATED)
+//    Candidate newCandidateFromIssue(@PathVariable String patternLanguageId, @RequestBody Issue issue) {
+//        logger.info(issue.getDescription());
+//        JSONObject contextObject = new JSONObject();
+//        contextObject.put("Context", issue.getDescription());
+//
+//        JSONArray contentArray = new JSONArray();
+//        contentArray.appendElement(contextObject);
+//
+//        JSONObject contentObject = new JSONObject();
+//        contentObject.put("content", contentArray);
+//
+//        Candidate candidate = new Candidate();
+//        candidate.setUri(issue.getName());
+//        candidate.setContent(contentObject);
+//        candidate.setName(issue.getName());
+//        candidate.setPatternLanguage(patternLanguageId.toString());
+////        return candidate;
+//        return this.candidateService.createCandidate(candidate);
+//    }
+
+    /**
+     * UPDATE Methods
+     */
+    @PutMapping(value = "/update")
+    Candidate putCandidate(@RequestBody Candidate candidate) {
+        logger.info(candidate.toString());
+        return this.candidateService.updateCandidate(candidate);
+    }
+
+    @PutMapping(value = "/updateRating/{candidateId}&{userId}&{rating}")
```

Comment:
stick to REST conventions: put on "/candidates/{candidateId}/rating/" - declare userId and rating as pathvariable, don't use '&' in the url mapping